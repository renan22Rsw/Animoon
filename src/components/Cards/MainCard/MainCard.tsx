import Image from "next/image";
import Link from "next/link";
import { usePathname } from "next/navigation";
import React from "react";

interface MainCardProps {
  title: string;
  images: string;
  id: number;
}

const MainCard = ({ title, images, id }: MainCardProps) => {
  const params = usePathname();
  const urlName: string[] = params?.split("/") || [];
  const url = urlName[2] || "default section";

  return (
    <div className=" py-2 px-1 lg:mx-4">
      <Link href={`/${url}/${id}`}>
        <Image
          className=" rounded-md cursor-pointer lg:hover:scale-110 h-[190px] lg:duration-700 lg:ease-in-out md:h-[265px]"
          src={images}
          width={200}
          height={0}
          priority={true}
          alt="card-picture"
        />
      </Link>

      <h2 className="py-4  text-xs md:text-sm  font-bold">
        {title.length > 35 ? title.substring(0, 35) + "..." : title}
      </h2>
    </div>
  );
};

export default MainCard;
