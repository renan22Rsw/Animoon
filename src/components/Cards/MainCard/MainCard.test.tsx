import { render, screen } from "@testing-library/react";
import MainCard from "./MainCard";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => ({
  usePathname: jest.fn(),
}));

describe("MainCard component", () => {
  it("should render animes route", () => {
    (usePathname as jest.Mock).mockReturnValue(`/some-path/animes`);
    render(
      <MainCard
        id={1}
        title="Naruto"
        images="https://example.com/anime-image.jpg"
      />
    );

    const titleElement = screen.getByText("Naruto");
    expect(titleElement).toBeInTheDocument();

    const linkElement = screen.getByRole("link");
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/animes/1");

    const imgElement = screen.getByRole("img");
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src");
    expect(imgElement).toHaveAttribute("width", "300");
    expect(imgElement).toHaveAttribute("height", "300");
  });

  it("should render mangas route", () => {
    (usePathname as jest.Mock).mockReturnValue(`/some-path/mangas`);
    render(
      <MainCard
        id={1}
        title="Naruto"
        images="https://example.com/manga-image.jpg"
      />
    );

    const titleElement = screen.getByText("Naruto");
    expect(titleElement).toBeInTheDocument();

    const linkElement = screen.getByRole("link");
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/mangas/1");

    const imgElement = screen.getByRole("img");
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src");
    expect(imgElement).toHaveAttribute("width", "300");
    expect(imgElement).toHaveAttribute("height", "300");
  });

  it("should render characters route", () => {
    (usePathname as jest.Mock).mockReturnValue(`/some-path/characters`);
    render(
      <MainCard
        id={1}
        title="Naruto"
        images="https://example.com/characters-image.jpg"
      />
    );
    const titleElement = screen.getByText("Naruto");
    expect(titleElement).toBeInTheDocument();

    const linkElement = screen.getByRole("link");
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/characters/1");

    const imgElement = screen.getByRole("img");
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src");
    expect(imgElement).toHaveAttribute("width", "300");
    expect(imgElement).toHaveAttribute("height", "300");
  });

  it("should match a snapshot", () => {
    const { container } = render(
      <MainCard id={1} title="Naruto" images="https://example.com/image.jpg" />
    );
    expect(container).toMatchSnapshot();
  });
});
