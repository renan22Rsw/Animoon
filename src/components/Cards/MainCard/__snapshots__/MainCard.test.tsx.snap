// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard component should match a snapshot 1`] = `
<div>
  <div
    class=" py-2 px-1 lg:mx-4"
  >
    <a
      href="/characters/1"
    >
      <img
        alt="card-picture"
        class=" rounded-md cursor-pointer lg:hover:scale-110 h-[190px] lg:duration-700 lg:ease-in-out md:h-[265px]"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="0"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </a>
    <h2
      class="py-4  text-xs md:text-sm  font-bold"
    >
      Naruto
    </h2>
  </div>
</div>
`;
